src/msxswagger/templates/swagger-sso-redirect.html

```diff
@@ -0,0 +1,125 @@
+<!doctype html>
+<html lang="en-US">
+<head><title>SSO Redirect</title></head>
+<body onload="run()">
+</body>
+</html>
+<script>
+    'use strict';
+    var InProgressRedirectKey = "CurrentRedirectSSO";
+    var SsoStateKeyPrefix = "sso-";
+
+    function run () {
+        var qp, arr;
+
+        if (/code|token|error/.test(window.location.hash)) {
+            qp = window.location.hash.substring(1);
+        } else {
+            qp = location.search.substring(1);
+        }
+
+        arr = qp.split("&")
+        arr.forEach(function (v,i,_arr) { _arr[i] = '"' + v.replace('=', '":"') + '"';})
+        qp = qp ? JSON.parse('{' + arr.join() + '}',
+            function (key, value) {
+                return key === "" ? value : decodeURIComponent(value)
+            }
+        ) : {}
+
+        var redirectCtx = loadRedirectContext({});
+        var sentState = redirectCtx.state;
+
+        var result = qp;
+        result.redirectUrl = redirectCtx.redirectUrl;
```

Comment:
better not to mutate `result` here.

We use the redirect url from `redirectCtx` down below anyway.

Instead of `result.isValid`, it’d have been less intrusive to use something like `const isValid = qs.state === sentState` and use it like `if(!isValid)` down below.

when we modify result, we are modifying `qp` (by reference -- result is not a clone, but it’s a direct reference to qp), and if the code grows in size, it might be harder to track and maintain.